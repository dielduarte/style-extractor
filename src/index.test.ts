import { expect, it, describe, vi } from 'vitest';

import { extractStyle } from './index';
import { Strategy } from './index.types';

const html = `
  <div style="width: 100%;border: red;">
    <style>
      h1 { line-height: 20px; }
    </style>
    <h1 style="color: white; font-size: 14px;">title</h1>  
  </div>
`;

const defaultOptions = {
  strategy: Strategy.scoped,
  classPrefix: 'es-',
};

describe('extractStyle => scoped', () => {
  it('should remove all inline styles', async () => {
    const result = await extractStyle(html, defaultOptions);
    expect(result.html).toMatchSnapshot();
  });

  it('should remove all inline style tags', async () => {
    const result = await extractStyle(html, defaultOptions);

    expect(result.html).toMatchSnapshot();
  });

  it('should generate scoped css', async () => {
    const result = await extractStyle(html, defaultOptions);

    expect(result.css).toMatchSnapshot();
  });
});

describe('extractStyle => atomic', () => {
  const options = { ...defaultOptions, strategy: Strategy.atomic };

  it('should remove all inline styles', async () => {
    const result = await extractStyle(html, options);
    expect(result.html).toMatchSnapshot();
  });

  it('should remove all inline style tags', async () => {
    const result = await extractStyle(html, options);

    expect(result.html).toMatchSnapshot();
  });

  it('should generate atomic css', async () => {
    const result = await extractStyle(html, options);

    expect(result.css).toMatchSnapshot();
  });
});
