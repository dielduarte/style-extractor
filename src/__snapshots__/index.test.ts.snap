// Vitest Snapshot v1

exports[`extractStyle => atomic > should generate atomic css 1`] = `
".es-cwtpWi { width:100%}
.es-gSvEpd { border:red}

      h1 { line-height: 20px; }
    .es-jTfyaK { color:white}
.es-dhZwgM { font-size:14px}
"
`;

exports[`extractStyle => atomic > should remove all inline style tags 1`] = `
"
  <div class=\\"es-cwtpWi es-gSvEpd \\">
    
    <h1 class=\\"es-jTfyaK es-dhZwgM \\">title</h1>  
  </div>
"
`;

exports[`extractStyle => atomic > should remove all inline styles 1`] = `
"
  <div class=\\"es-cwtpWi es-gSvEpd \\">
    
    <h1 class=\\"es-jTfyaK es-dhZwgM \\">title</h1>  
  </div>
"
`;

exports[`extractStyle => scoped > should generate scoped css 1`] = `
".es-chrTOf { width: 100%;border: red; }

      h1 { line-height: 20px; }
    .es-dfoHTq { color: white; font-size: 14px; }
"
`;

exports[`extractStyle => scoped > should remove all inline style tags 1`] = `
"
  <div class=\\"es-chrTOf \\">
    
    <h1 class=\\"es-dfoHTq \\">title</h1>  
  </div>
"
`;

exports[`extractStyle => scoped > should remove all inline styles 1`] = `
"
  <div class=\\"es-chrTOf \\">
    
    <h1 class=\\"es-dfoHTq \\">title</h1>  
  </div>
"
`;
